src/routing_policy.rs

```diff
@@ -0,0 +1,79 @@
+use std::sync::Arc;
+
+use murmur3::murmur3_32;
+
+use crate::producer::Message;
+
+#[derive(Clone, Default)]
+pub enum RoutingPolicy {
+    #[default]
+    RoundRobin,
+    Single(usize),
```

Comment:
This strategy is different from other clients. The index should be generated once rather than provided by the application side. Because the application cannot know the actual number of partitions in advance, the index could exceed the actual number of partitions